src/transpilation/resolve.ts

```diff
@@ -184,17 +186,35 @@ class ResolutionContext {
         // We don't know for sure where the lua root is, so guess it is at package root
         const splitPath = path.normalize(requiringFile.fileName).split(path.sep);
         let packageRootIndex = splitPath.lastIndexOf("node_modules") + 2;
-        let packageRoot = splitPath.slice(0, packageRootIndex).join(path.sep);
+        const packageRoot = splitPath.slice(0, packageRootIndex).join(path.sep);
+        let currentPackage = packageRoot;
 
         while (packageRootIndex < splitPath.length) {
             // Try to find lua file relative to currently guessed Lua root
-            const resolvedPath = path.join(packageRoot, dependency);
+            const resolvedPath = path.join(currentPackage, dependency);
             const fileFromPath = this.getFileFromPath(resolvedPath);
             if (fileFromPath) {
                 return fileFromPath;
             } else {
                 // Did not find file at current root, try again one directory deeper
-                packageRoot = path.join(packageRoot, splitPath[packageRootIndex++]);
+                currentPackage = path.join(currentPackage, splitPath[packageRootIndex++]);
+            }
+        }
+
+        if (this.plugins != null) {
```

Comment:
I don't think this should be exclusive to nodeModules paths, I think a nicer place for this would be in `resolveImport`, before the call to `resolveDependencyPath` (going with the idea I mentioned on the plugin interface above, where users can intercept these calls before we do all of this quite complicated work on the file system while we already know we won't find anything)